features/networking/metrics.feature

```diff
@@ -0,0 +1,60 @@
+Feature: SDN/OVN metrics related networking scenarios
+
+  # @author anusaxen@redhat.com
+  # @case_id OCP-28519
+  @admin
+  Scenario: Prometheus should be able to monitor kubeproxy metrics
+    Given I switch to cluster admin pseudo user
+    And I use the "openshift-sdn" project
+    When I run the :get client command with:
+      | resource | ep            |
+      | n        | openshift-sdn |
+      | o        | json          |
+    And evaluation of `@result[:parsed]['items'][0]['subsets'][0]['addresses'][1]['ip'].to_s` is stored in the :metrics_ep_ip clipboard
+    And evaluation of `@result[:parsed]['items'][0]['subsets'][0]['ports'][0]['port'].to_s` is stored in the :metrics_ep_port clipboard
```

Comment:
@anuragthehatter there's an `endpoint` call that you can use to avoid using the raw `:get` method.  For example
```
[12] pry(#<BushSlicer::DefaultWorld>)> endpoints('sdn-metrics').subsets.first.addresses.first
=> #<BushSlicer::EndpointAddress:0x00007ffb319220c8
 @struct=
  {"ip"=>"10.0.128.172",
   "nodeName"=>"ip-10-0-128-172.us-east-2.compute.internal",
   "targetRef"=>
    {"kind"=>"Pod",
     "name"=>"sdn-metrics-knvpd",
     "namespace"=>"openshift-sdn",
     "resourceVersion"=>"40581",
     "uid"=>"e323a796-d871-41d8-89d5-a057af517796"}}>
```
```
[17] pry(#<BushSlicer::DefaultWorld>)> endpoints('sdn-metrics').subsets.first.ports.first.port
=> 9101```

Reply:
Ya the prob is there is a change in endpoint name across various releases and ```endpoints('<name>')``` expects a static name so had to use it

Reply:
What the differences?  Are the resource labels the same.  For 4.6 I see the following

```$ oc get endpoints -n openshift-sdn --show-labels
NAME          ENDPOINTS                                                          AGE   LABELS
sdn-metrics   10.0.128.172:9101,10.0.137.241:9101,10.0.140.99:9101 + 8 more...   9h    app=sdn-metrics,service.kubernetes.io/headless=```

Reply:
Yea, for  <4.4, label is ```app=sdn``` and ep name is ```sdn```

Reply:
@anuragthehatter we can construct the label to adapt according to the ocp version.  Something like the following
```    And evaluation of `env.version_le("4.4", user: user) ? "sdn" : "sdn-metrics"` is stored in the :sdn_label clipboard
    And evaluation of `endpoints(cb.sdn_label).subsets.first.addresses.first.ip.to_s` is stored in the :metrics_ep_ip clipboard
    And evaluation of `endpoints(cb.sdn_label).subsets.first.ports.first.port.to_s` is stored in the :metrics_ep_port clipboard```

Reply:
@zhaozhanqi...you can leverage the same logic for your PR #1396

Reply:
Gonna give that a try @pruan-rht . Thanks for suggestions meanwhile.

Reply:
wow, ruby is kinda expressive

```ruby
    And evaluation of `BushSlicer::Endpoints.list(user: user, project: project).select {|e| e.name =~ /sdn|sdn-metrics/}.first.subsets.first` is stored in the :metrics_ep clipboard
    And evaluation of `cb.metrics_ep.addresses.first.ip.to_s` is stored in the :metrics_ep_ip clipboard
    And evaluation of `cb.metrics_ep.ports.first.port.to_s` is stored in the :metrics_ep_port clipboard
```

probably too much though, but a fun exercise

Reply:
Pheww. this was bit complicated to test. Had to change things a bit but your suggestions really helped, Peter. Thanks. @rbbratta  Yea seems like these fun exercises might be good for me on learning friday 

Ok guys so if version is <4.6 we expect metrics to use clear traffic means over http and if >=4.6 it has to be https as suggested in PR #1396 so modifed accordingly to interoperate btw various releases

4.6 logs for both tests: https://privatebin-it-iso.int.open.paas.redhat.com/?917e3894be6c6db5#w4h5wImSR9v4xcBdilsVoTaD9ozIDvW3KbwuVlvAJOg=

4.5 logs for both tests: https://privatebin-it-iso.int.open.paas.redhat.com/?d9c22450d6b76169#ie/fhnKiM068+WGXrdJ1WQdRcvc2MqicyOyHmE862UQ=

@pruan-rht Just to correct me and you, the metrics name is ```sdn``` until 4.5 and ```sdn-metrics``` staring 4.6.